README.md

```diff
@@ -134,10 +136,46 @@ To visualize the pipeline, follow these steps:
 
 **Note** For linux-based users you can directly launch the pipeline visualizer with the option ```-gui``` option in simulate script
 
+---
+## Installation with Docker
+
+### Prerequisites
+- [Docker](https://docs.docker.com/get-docker/)
+- A display server to attach to the `konata` container. For macOS, it is recommended to use [XQuartz](https://gist.github.com/sorny/969fe55d85c9b0035b0109a31cbcb088).
+
+### Starting the Environment
+To start the Docker environment, use:
+```bash
+docker compose up -d
+```
+
+### Usage
+The setup includes the gem5 simulator and a RISC-V cross-compiler within the `riscv` container. You can attach to this container and run simulations as needed. Simulation results are saved in the `results` directory, and you can view the generated `.out` files using the Konata visualization tool.
+
+Konata is available in the `konata` container and can be run independently from the gem5 simulation.
+
+### Running the RISC-V Cross-Compiler
```

Comment:
this is incorrect!
the cross compiler is a part of the flow! please read carefully the README.md and the slides.